src/rust/protocol/src/memcache/wire/response/mod.rs

```diff
@@ -19,53 +26,259 @@ pub enum MemcacheResponse {
     Stored,
 }
 
+impl MemcacheResult {
+    fn len(&self) -> usize {
+        self.as_bytes().len()
+    }
+
+    fn as_bytes(&self) -> &[u8] {
+        match self {
+            Self::Deleted => b"DELETED\r\n",
+            Self::Exists => b"EXISTS\r\n",
+            Self::Values { .. } => b"VALUE ",
+            Self::NotFound => b"NOT_FOUND\r\n",
+            Self::NotStored => b"NOT_STORED\r\n",
+            Self::Stored => b"STORED\r\n",
+        }
+    }
+
+    fn code(&self) -> usize {
+        match self {
+            Self::Deleted => 7,
+            Self::Exists => 6,
+            Self::NotFound => 8,
+            Self::NotStored => 9,
+            Self::Stored => 9,
```

Comment:
this should be 5?

Reply:
Since this is done for backward compatibility, maybe add a reference to https://github.com/twitter/pelikan/blob/master/src/protocol/data/memcache/response.h